### ISSUE TITLE
Add back `tuple_pattern` rule?

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#3466]
The rule was removed in https://github.com/realm/SwiftLint/commit/ed37c284197ab4597b588b891655cdee6982938a, but never added back again.

**Comment by sindresorhus** [comment]
Actually, this applies to all the rules that were removed in https://github.com/realm/SwiftLint/releases/tag/0.39.1

**Comment by sindresorhus** [comment]
Please keep this open.

**Comment by jpsim** [comment]
We never solved how to package releases with SwiftSyntax in a safe and portable way. We need to solve that before re-adding this rule.

**Comment by marcelofabri** [comment]
Closing this because this is now a warning in the compiler, so the only advantage here would be warning people using older Swift versions. But my gut feeling is that it's not worth maintaining another rule just for that.

```
$ swift file.swift       
file.swift:1:20: warning: expression shuffles the elements of this tuple; this behavior is deprecated
let (x: a, y: b) = (y: 4, x: 5.0)
```

This warning was added in Swift 5.3 (https://github.com/apple/swift/commit/a9e871a0b13e6233b532ae74f843ebf5e7d13747) which is ~1.5 years old.
